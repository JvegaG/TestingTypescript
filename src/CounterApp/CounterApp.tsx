import React, { useEffect, useState } from 'react'
import { getImagen } from '../base-pruebas/11-async-await';

interface counterProps {
    value: number
}

const CounterApp: React.FC<counterProps> = ({ value }) => {


    const [newValue, setNewValue] = useState<number>(value);
    const [newImage, setNewImage] = useState<string>('');

    const handleSum = () => {
        setNewValue(newValue + 1)
    }

    const handleRest = () => {
        setNewValue(newValue - 1)
    }

    const handleReset = () => {
        setNewValue(value)
    }

    const fetchImage = async () => {
        const src = await getImagen();

        setNewImage(src)
    }
    
    useEffect(() => {
        fetchImage();
    }, [])

    return (
        <>
            <h1 style={{ fontWeight: 'bold' }}>CounterApp</h1>
            <h2>{newValue}</h2>

            <div className="mb-2">
                <img src={newImage}></img>
            </div>

            <button
                className="btn btn-primary d-inline me-2"
                onClick={handleSum}
            > +1 </button>

            <button
                className="btn btn-primary d-inline me-2"
                onClick={handleRest}
            > -1 </button>

            <button
                className="btn btn-primary d-inline"
                onClick={handleReset}
            > Reset </button>
        </>
    )
}

export default CounterApp