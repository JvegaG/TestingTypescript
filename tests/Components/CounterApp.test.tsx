import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react'
import CounterApp from '../../src/Components/CounterApp'


describe('test component <CounterApp>', () => {

    const initialValue = 100;

    test('should create a snapshot', () => {

        const { container } = render(<CounterApp value={initialValue} />);

        expect(container).toMatchSnapshot()

    })

    test('should show 100 like initial counter', () => {

        render(<CounterApp value={initialValue} />);

        expect(screen.getByText(initialValue)).toBeTruthy();
        expect(screen.getByText(initialValue).innerHTML).toBe(initialValue.toString());
        expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain(initialValue.toString());
    })


    test('should increment +1 click button +1', () => {

        render(<CounterApp value={initialValue} />);

        fireEvent.click(screen.getByText('+1')) // Simula el evento click del button '+1'

        expect(screen.getByText(initialValue + 1)).toBeTruthy();

    })


    test('should decrement -1 click button -1', () => {

        render(<CounterApp value={initialValue} />);

        fireEvent.click(screen.getByText('-1')) // Simula el evento click del button '-1'
        // screen.debug()

        expect(screen.getByText(initialValue - 1)).toBeTruthy();

    })

    test('should do the event from reset button', () => {

        render(<CounterApp value={initialValue} />);

        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByText('+1'))
        // fireEvent.click(screen.getByText('Reset'));
        fireEvent.click(screen.getByRole('button',{name:'btn-reset'}))
        

        expect(screen.getByText(initialValue)).toBeTruthy();

    })
})