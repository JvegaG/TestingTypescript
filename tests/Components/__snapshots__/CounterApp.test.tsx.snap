// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component <CounterApp> should create a snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
    style="font-weight: bold;"
  >
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <div
    class="mb-2"
  >
    <img
      src=""
    />
  </div>
  <button
    class="btn btn-primary d-inline me-2"
  >
     +1 
  </button>
  <button
    class="btn btn-primary d-inline me-2"
  >
     -1 
  </button>
  <button
    aria-label="btn-reset"
    class="btn btn-primary d-inline"
  >
     Reset 
  </button>
</div>
`;
